src/nimony/sem.nim

```diff
@@ -4949,6 +4949,14 @@ proc semcheck*(infile, outfile: string; config: sink NifConfig; moduleFlags: set
       c.dest.copyTree res.decl
   instantiateGenericHooks c
   wantParRi c, n
+
+  if reportErrors(c) == 0:
+    var final = move c.dest
+    var finalBuf = beginRead(final)
+    c.dest = injectDerefs(finalBuf)
```

Comment:
You need to report errors for this one as well... right?

Reply:
Yeah, It is. Errors from `deref` should be caught by the `reportErrors` below. `derefs` doesn't work well with erroneous code from the preceding phase. This stops injecting derefs if there are already errors.